src/contexts/Global/index.tsx

```diff
@@ -0,0 +1,294 @@
+   * The cookie bar is the bar that appears when the user enters the website
+   * for the first time.
+   *
+   * @note This value is set to `true` after a delay of `COOKIE_BAR_ENTER_DELAY`,
+   * provided the user has not previously dismissed the cookie bar.
+   * @default false
+   */
+  isCookieBarOpen: boolean
+  /**
+   * A `boolean` which indicates whether the market menu is open or not.
+   * The market menu contains the list of markets that the user can switch between.
+   * @default false
+   */
+  isMarketMenuOpen: boolean
+  /**
+   * A `boolean` which indicates whether the navigation menu is open or not.
+   * The navigation menu is the menu that appears when the user clicks on the
+   * three stacked lines icon, or otherwise known as the _hamburger_ icon.
+   * @default false
+   */
+  isNavMenuOpen: boolean
+  /**
+   * A `boolean` which indicates whether the search menu is open or not.
+   * The search menu is the menu that appears when the user clicks on the search
+   * icon.
+   * @default false
+   */
+  isSearchMenuOpen: boolean
+  /**
+   * A `boolean` which indicates whether the store message is open or not.
+   * The store message is usually located above the site header and is used to
+   * display important messages to the user.
+   * @default false
+   */
+  isStoreMessageOpen: boolean
+  /**
+   * A `boolean` which indicates whether any of the following menu's are open:
+   * - `isCartMenuOpen`
+   * - `isNavMenuOpen`
+   * - `isSearchMenuOpen`
+   * @default false
+   */
+  isSomeMenuOpen: boolean
+}
+
+type GlobalStateCTX = GlobalState | undefined
+export const GlobalStateContext = React.createContext<GlobalStateCTX>(undefined)
+
+interface GlobalHandlers {
+  /**
+   * A setter function which sets the `isCartMenuOpen` state to `false`.
+   */
+  onCartMenuClose: () => void
+  /**
+   * A setter function which toggles the `isCartMenuOpen` state.
+   * @note This function also sets the `isNavMenuOpen` and `isSearchMenuOpen`
+   * states to `false`.
+   */
+  onCartMenuToggle: () => void
+  /**
+   * A setter function which sets the `isCookieBarOpen` state to `false`.
+   */
+  onCookieBarClose: () => void
+  /**
+   * A setter function which sets the `isMarketMenuOpen` state to `false`.
+   */
+  onMarketMenuClose: () => void
+  /**
+   * A setter function which toggles the `isMarketMenuOpen` `boolean` state.
+   */
+  onMarketMenuToggle: () => void
+  /**
+   * A setter function which sets the `isNavMenuOpen` state to `false`.
+   */
+  onNavMenuClose: () => void
+  /**
+   * A setter function which toggles the `isNavMenuOpen` state.
+   *
+   * @note This function also sets the `isCartMenuOpen` and `isSearchMenuOpen`
+   * states to `false`.
+   */
+  onNavMenuToggle: () => void
+  /**
+   * A setter function which sets the `isSearchMenuOpen` state to `false`.
+   *
+   * @note This function also sets the `isNavMenuOpen` and `isCartMenuOpen`
+   * states to `false`.
+   */
```

Comment:
> @note This function also sets the `isNavMenuOpen` and `isCartMenuOpen`
> states to `false`.

Wonder if this prone for becoming invalid as well. 🤔

Reply:
Agreed. We should look this over and discuss. I simply added comments to reflect the existing logic.